server/src/main/java/org/elasticsearch/index/mapper/BinaryRangeUtil.java

```diff
@@ -69,6 +106,43 @@ static BytesRef encodeDoubleRanges(Set<RangeFieldMapper.Range> ranges) throws IO
         return new BytesRef(encoded, 0, out.getPosition());
     }
 
+    static List<RangeFieldMapper.Range> decodeDoubleRanges(BytesRef encodedRanges) {
+        return decodeRanges(encodedRanges, RangeFieldMapper.RangeType.DOUBLE,
+            BinaryRangeUtil::decodeDouble);
+    }
+
+    static List<RangeFieldMapper.Range> decodeFloatRanges(BytesRef encodedRanges) {
+        return decodeRanges(encodedRanges, RangeFieldMapper.RangeType.FLOAT,
+            BinaryRangeUtil::decodeFloat);
+    }
+
+    static List<RangeFieldMapper.Range> decodeRanges(BytesRef encodedRanges, RangeFieldMapper.RangeType rangeType,
+                                                     TriFunction<byte[], Integer, Integer, Object> decodeBytes) {
+
+        BinaryDocValuesRangeQuery.LengthType lengthType = rangeType.lengthType;
```

Comment:
Nit: It seems a bit weird for a utility class to be using something from a query class rather than the other way around. Maybe we should consider either moving `BinaryDocValuesRangeQuery.LengthType` to be an inner class of this `BinaryRangeUtil` class or make it a standalone class on its own?

Reply:
Or even make it an inner class of `RangeFieldMapper`?

Reply:
I kind of feel like we already have too many inner classes crammed into `RangeFieldMapper`, but moving `LengthType` to `BinaryRangeUtil` makes a lot of sense to me.  Pushed a change set doing that :)